### PR TITLE
warcio/cli.py Add extract command to print specified WARC records.

### PR DESCRIPTION
**Pull request by adam-miller**

### CODE REVIEW

File: warcio/cli.py [hunk 1]
```diff
@@ -93,6 +105,22 @@ def decompress_and_recompress(self, stream, output):
             tout.seek(0)
             self.load_and_write(tout, output)
 
+# ============================================================================
+def extract_record(cmd):
+    records = cmd.records.split(',')
+    writer = WARCWriter(filebuf=sys.stdout.buffer, gzip=False)
+    for filename in cmd.inputs:
+        with open(filename, 'rb') as fh:
+            for record in ArchiveIterator(fh, no_record_parse=True, arc2warc=True):
```
Comment (ikreymer) [inline warcio/cli.py]: Do you want to have `arc2warc` be true? If it is true, the ARC record is converted on the fly to an WARC record, so only the 2nd condition would be needed. But, you may want to print the original ARC record here when extracting and not convert.

**Comment by ikreymer** [comment]
Hey Adam, thanks for the PR!
Do you mind adding some quick tests? You can use one of the existing WARCs or ARCs in `./test/data`

**Comment by adam-miller** [comment]
@nlevitt suggested renaming the verb to "filter" as well, which makes a little more sense given the iterative behavior. So, I'll do a little refactoring and add some tests.

**Comment by ikreymer** [comment]
Closing as this was superseded by #26

**Pull request closed without merging**